src/stirling/utils/linux_headers.cc

```diff
@@ -229,7 +229,20 @@ StatusOr<std::filesystem::path> ResolvePossibleSymlinkToHostPath(const std::file
     return error::Internal(ec.message());
   }
 
-  const auto resolved_host_path = system::Config::GetInstance().ToHostPath(resolved);
+  // Relative paths containing "../" can result in an invalid host mount path when using
+  // ToHostPath. Therefore, we need to treat the absolute and relative cases differently.
+  std::filesystem::path resolved_host_path = p.parent_path();
+  if (resolved.is_absolute()) {
+    resolved_host_path = system::Config::GetInstance().ToHostPath(resolved);
+    LOG(INFO) << absl::Substitute(
+        "Symlink target is an absolute path. Converting that to host path: $0 -> $1.",
+        resolved.string(), resolved_host_path.string());
+  } else {
+    resolved_host_path /= resolved.string();
+    LOG(INFO) << absl::Substitute(
+        "Symlink target is a relative path. Concatenating it to parent directory: $0",
+        resolved_host_path.string());
+  }
```

Comment:
Maybe convert the LOGs to VLOGs (or remove them).

Reply:
Fixed in ed7b839d630c0ee2acdde1319a008d6d099e0e18